File: lab_2/task1/PRNG.cpp
#include <iostream>
#include <random>
#include <string>
#include <fstream>

void GenRandomSeq(const std::string& path) {
    std::mt19937 generator(std::random_device{}());
    int min = 0;
    int max = 1;
    std::uniform_int_distribution<int> distribution(min, max);
    std::string seq;
    for (int i = 0; i < 128; ++i) {
        seq+= std::to_string(distribution(generator));
    }
    std::ofstream file(path);
	file << seq;
}

int main(){
    GenRandomSeq("Random_sequence_cpp.txt");
}

File: lab_2/task1/PRNG.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class PRNG {
    public static void genRandomSeq(String path) {
        Random random = new Random();
        StringBuilder seq = new StringBuilder();
        for (int i = 0; i < 128; i++) {
            seq.append(random.nextInt(2));
        }
        try (FileWriter fileWriter = new FileWriter(path)) {
            fileWriter.write(seq.toString());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    public static void main(String[] args) {
        genRandomSeq("Random_sequence_java.txt");
    }
}